frontend/src/data/variables/VariableProvider.ts

```diff
@@ -1,28 +1,17 @@
 import { Breakdowns } from "../Breakdowns";
-import { Dataset } from "../DatasetTypes";
 import { ProviderId, MetricId } from "../variableProviders";
 import { MetricQueryResponse, createMissingDataResponse } from "../MetricQuery";
 
 abstract class VariableProvider {
   readonly providerId: ProviderId;
   readonly providesMetrics: MetricId[];
-  readonly datasetIds: readonly string[];
 
-  constructor(
-    providerId: ProviderId,
-    providesMetrics: MetricId[],
-    datasetIds: string[]
-  ) {
+  constructor(providerId: ProviderId, providesMetrics: MetricId[]) {
     this.providerId = providerId;
     this.providesMetrics = providesMetrics;
-    this.datasetIds = datasetIds;
   }
 
-  // TODO change return type to MetricQueryResponse instead of Row[]
-  getData(
-    datasets: Record<string, Dataset>,
-    breakdowns: Breakdowns
-  ): MetricQueryResponse {
+  async getData(breakdowns: Breakdowns): Promise<MetricQueryResponse> {
```

Comment:
Do the tests all pass without any changes now that this is async?

Reply:
Sorry, was still working on tests. They should be passing now! (Also the variable provider tests are somewhat simpler)